src/controllers/client.js

```diff
@@ -117,7 +117,7 @@ async function assignEchisClientUPI(req, res) {
     success(201, res.locals.echisClient._id, clientNumber)
   } catch (err) {
     const errStruct = { instance: getCHTValuesFromEnv(instance).url, user: res.locals.echisClient._id, error: err.message };
-    const errString = `could not complete creating client ${JSON.stringify(errStruct)})`
+    const errString = `Could not complete creating client. Possible Missing Fields. ${JSON.stringify(errStruct)})`
```

Comment:
`Possible Missing Fields` isn't the only reason the request could fail. The error `400` returned by the client registry API should serve as a hint for the reason. Does the API return the missing fields if there are some? If so then maybe using the error from CR would be more helpful

Reply:
The issue is some of the functions do not have a catch statement. Such as the createCRClient Function. https://github.com/moh-kenya/echis-2.0-integration/blob/512756642206300ad616c088808822361c5de3c2/src/controllers/client.js#L45

```
const createCRClient = (conf, echisClient) => {
  const payload = generateClientRegistryPayload(echisClient);
  return axiosInstance.post(`partners/registry`, payload, {
    headers: {
      "Content-Type": "application/json",
    },
  }).then((resp) => {
    return updateDoc(conf, echisClient._id, { upi: resp.data.clientNumber })
      .then(() => Promise.resolve(resp.data.clientNumber));
  }).catch((err)=>{
      logger.error(err.response.data.errors);
  })
}
```

Possiby as above and handling appropriately

Reply:
> The issue is some of the functions do not have a catch statement.

That's by design, the caller of the function should catch the error. If the CR errors are in `err.response.data.errors`, can't you just use that where the function is called?

Reply:
Reverted the changes.